lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox { ...minimalProps } autoComplete={true}/>);
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const { getByRole, rerender } = quickRender({
+              autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }]
+            }),
+            inputElement = getByRole('combobox');
+
+        await userEvent.type(inputElement, 'f');
+        rerender(
+          <NxCombobox { ...minimalProps }
+                      autoComplete={true}
+                      matches={ [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }/>);
+        expect(inputElement).toHaveValue('Foo');
+      });
+
+  it('sets aria-expanded on the input to true when the dropdown displayed which has aria-hidden set to false',
+      async function() {
+        const { getByRole } = quickRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+            inputElement = getByRole('combobox'),
+            dropdownElement = getByRole('listbox', { hidden: true });
+
+        expect(inputElement).toHaveAttribute('aria-expanded', 'false');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'true');
+        await userEvent.type(inputElement, 'f');
+        expect(inputElement).toHaveAttribute('aria-expanded', 'true');
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'false');
+      });
+
+  it('calls onChange whenver the input\'s onChange event fires', async function() {
+    const onChange = jest.fn(),
+        inputElement = quickRender({ onChange }).getByRole('combobox');
+
+    expect(onChange).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'a');
+    expect(onChange).toHaveBeenCalledWith('a');
+  });
+
+  it('does not call onSearch whenever the input\'s onChange event fires with a value that does not differ' +
+    'after trimming', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = quickRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, ' ');
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('calls onSearch whenever the input\'s onChange event fires with a value that differs after trimming, ' +
+    'passing the trimmed value', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = quickRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'f');
+    expect(onSearch).toHaveBeenCalledWith('foof');
+  });
+
+  it('passes the disabled prop to the input and buttons', function() {
```

Comment:
This test doesn't check the buttons.  Do the buttons really even need to be disabled? Or should the whole dropdown just not be present for disabled comboboxes?

Reply:
Since we have `aria-hidden` applied to the dropdown which removes itself and all of its children from the accessibility tree and the tabindex="-1" will make sure the button are not in the keyboard tabbing order. I think we don't need the disabled attribute on button in this case.

Reply:
> Since we have aria-hidden applied to the dropdown which removes itself and all of its children from the accessibility tree and the tabindex="-1" will make sure the button are not in the keyboard tabbing order. I think we don't need the disabled attribute on button in this case.

In that case the test name should be adjusted

Reply:
Updated here 6dbc601488edec3acdf06ca1c289688de300a1a8.